query-planner-js/src/buildPlan.ts

```diff
@@ -969,20 +1140,46 @@ function sameMergeAt(m1: ResponsePath | undefined, m2: ResponsePath | undefined)
   return arrayEquals(m1, m2);
 }
 
+/**
+ * Used in `FetchDependencyGraph` to store, for a given group, information about one of its parent.
+ * Namely, this structure stores:
+ *  1. the actual parent group, and
+ *  2. the path of the group for which this is a "parent relation" into said parent (`group`). This information
+ *   is maintained for the case where we want/need to merge groups into each other. One can roughly think of
+ *   this as similar to a `mergeAt`, but that is relative to the start of `group`. 
+ */
+type ParentRelation = {
+  group: FetchGroup,
+  path?: OperationPath,
+}
+
+function concatPathsInParents(first: OperationPath | undefined, second: OperationPath | undefined): OperationPath | undefined  {
+  return first && second ? concatOperationPaths(first, second) : undefined;
```

Comment:
If `first` is defined, but `second` is undefined, you want to return undefined and not `first`?

Reply:
Correct. `undefined` for "path in parents" means that it's "unknown and don't make any assumption on it", not that it's empty. I tried to specify that a bit better in the comment about `path` in `ParentRelation`.